kedro/framework/cli/starters.py

```diff
@@ -560,6 +560,7 @@ def _get_extra_context(  # noqa: PLR0913
         extra_context["example_pipeline"] = str(_parse_yes_no_to_bool(example_pipeline))
 
     # set defaults for required fields, will be used mostly for starters
+    # project_name defaut value will be taken from cookiecutter.json file
```

Comment:
The name will come from `cookiecutter.json` or from the user input right?

Reply:
Yes, user can provide input from prompt or config file, if not - default value from cookiecutter.json will be used

Reply:
To add more context, the project name **must** be provided by the user in one way or another - if omitted from provided config the project creation will fail. If not provided by flag and no config is given, then the user will be required to provide it in the user prompts. The value from cookiecutter.json is used if the user hits enter for this prompt, using the default value that is shown as part of the prompt (currently New Kedro Project) - but this will never be populated directly (or invisibly, rather) from cookiecutter.json

Reply:
Yes, I agree with @AhdraMeraliQB this isn't actually a "default" value. I feel like this comment creates more confusion than clarification so I'd probably vote to close the PR and not merge it.